src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector256.cs

```diff
@@ -1150,7 +1151,7 @@ internal static Vector256<T> Create<T>(Vector128<T> lower, Vector128<T> upper)
         /// <returns>A new <see cref="Vector256{T}" /> instance with the first element initialized to <paramref name="value" /> and the remaining elements initialized to zero.</returns>
         /// <exception cref="NotSupportedException">The type of <paramref name="value" /> (<typeparamref name="T" />) is not supported.</exception>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal static Vector256<T> CreateScalar<T>(T value)
+        public static Vector256<T> CreateScalar<T>(T value)
```

Comment:
Could we change this to

```c#
[MethodImpl(MethodImplOptions.AggressiveInlining)]
public static Vector256<T> CreateScalar<T>(T value)
    where T : struct
{
    if (Avx.IsSupported)
    {
         return Vector128.CreateScalar(value).ToVector256Unsafe();
    }

    return Vector128.CreateScalar(value).ToVector256();
}
```

I end up not using this method a lot of times because it currently emits an extra `vmovaps` to clear the upper lane after the scalar move, even though the scalar move will have already zeroed the upper lane with VEX encoding.

The extra `vmovaps` should get no-op'ed by the CPU front-end anyway, but this is a common need, so smaller codegen would be better.

Reply:
I'm going to update this to be a proper intrinsic in a follow up PR.